include/libdnf5/rpm/arch.hpp

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF5_RPM_ARCH_HPP
+#define LIBDNF5_RPM_ARCH_HPP
+
+#include <string>
+#include <vector>
+
+
+namespace libdnf5::rpm {
+
+std::vector<std::string> get_supported_arches();
```

Comment:
May be we need it, but I am confused another thing - structure from which the vector is generated.

It is c structure and it is not nice for our usage. The construct is <basearch>, <arches> therefore you have to iterate over the whole structure. If we would like to improve the search then we can create std::unordered_map<string, string> where key will be <arch> and the value <basearch>. The problem of this approach is that map is bigger then the original structure, and <Basearch> strings will be present multiple times, but it can be improved by another approach
```
std::unordered_map<string, int>  // <arch>, <index>
std::array<std::string>, where index could be used to retrieve the Basearch.
```
@m-blaha What do you think? @jrohel May I ask you for your opinion?

Reply:
It's used by dnf5 to generate an error message if unsupported arch is used.

Reply:
Regarding the `ARCH_MAP` data structure, you are right. I guess it was copied from libdnf where probably the C structure was needed. In dnf5 we can replace it by C++ structure (like std::unordered_map).

Reply:
may be `std::map` could be better if you want to have get_supported_arches ordered without ordering.

Reply:
- I would prefer `std::map` over `std::unordered_map` for these tiny maps.
- And inside `std::string_view`. No need to copy those strings to the heap.

But it will still take up more memory than the original code.

Reply:
@j-mracek do you want me to change also the underlying `ARCH_MAP` structure, or is TODO sufficient for this PR?

Reply:
@m-blaha Why do you actually want to replace the C structure? How many times do we call `get_base_arch`? Is it worth it to construct a map?
The cost to create a map will be several times higher than calling `get_base_arch`.

Is the requirement or do we need to intensively call `get_base_arch`? (Note: If so, then code can be written to create a sorted reverse mapping (array of native -> base structs) with little cost directly at compile time.)

Reply:
I would like to clarify internal stricture, because there is a method (public) which depends on. The method returns a vector with strings.

Reply:
I am fine even with TODO.

Reply:
@jrohel - I was thinking that the map is created during compilation. Or did I overlooked something?